src/AzureSqlAction.ts

```diff
@@ -90,15 +88,50 @@ export default class AzureSqlAction {
 
     private async _executeSqlFile(inputs: ISqlActionInputs) {
         core.debug('Begin executing sql script');
-        let scriptContents: string;
-        try {
-            scriptContents = fs.readFileSync(inputs.sqlFile, "utf8");
+
+        // sqlcmd should be added to PATH already, we just need to see if need to add ".exe" for Windows
+        let sqlCmdPath: string;
+        switch (process.platform) {
+            case "win32": 
+                sqlCmdPath = "sqlcmd.exe";
+                break;
+            case "linux":
+            case "darwin":
+                sqlCmdPath = "sqlcmd";
+                break;
+            default:
+                throw new Error(`Platform ${process.platform} is not supported.`);
         }
-        catch (e) {
-            throw new Error(`Cannot read contents of file ${inputs.sqlFile} due to error '${e.message}'.`);
+
+        // Determine the correct sqlcmd arguments based on the auth type in connectionConfig
+        let sqlcmdCall = `"${sqlCmdPath}" -S ${inputs.serverName} -d ${inputs.connectionConfig.Config.database}`;
+        const authentication = inputs.connectionConfig.Config['authentication'];
+        switch (authentication?.type) {
+            case undefined:
+                // No authentication type defaults SQL login
+                sqlcmdCall += ` -U "${inputs.connectionConfig.Config.user}"`;
+                core.exportVariable(Constants.sqlcmdPasswordEnvVarName, inputs.connectionConfig.Config.password);
+                break;
+
+            case 'azure-active-directory-default':
+                sqlcmdCall += ` --authentication-method=ActiveDirectoryDefault`;
+                break;
+
```

Comment:
for AD default does the user know they need to set the various environment variables like AZURE_CLIENT_ID and AZURE_TENANT_ID?

Reply:
I'll update the README for documentation